lib/puppet.rb

```diff
@@ -74,7 +74,7 @@ class << self
       end
 
       if locale_path
-        if Gem.loaded_specs['gettext-setup'].version < Gem::Version.new('0.8')
+        if GettextSetup.method(:initialize).parameters.count > 1
```

Comment:
This might need to be the opposite? As in, if it's 2 (or greater than 1), use the version that passes 2 arguments, otherwise just pass one.